llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -1398,6 +1399,8 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
     if (instrumentWithCalls(ConvertedShadow) &&
         SizeIndex < kNumberOfAccessSizes && !MS.CompileKernel) {
       FunctionCallee Fn = MS.MaybeWarningFn[SizeIndex];
+      // ZExt cannot convert between vector and scalar
+      ConvertedShadow = convertShadowToScalar(ConvertedShadow, IRB);
```

Comment:
we need IR test

Reply:
Baseline IR tests at https://github.com/llvm/llvm-project/pull/97611

I'll update this PR to show the corrected IR output